packages/dev/core/src/Materials/PBR/pbrBaseMaterial.ts

```diff
@@ -1253,6 +1253,15 @@ export abstract class PBRBaseMaterial extends PushMaterial {
             return false;
         }
 
+        // Check if Area Lights have LTC texture.
+        if (defines["AREALIGHTUSED"]) {
+            for (let index = 0; index < scene.lights.length; index++) {
+                if (!scene.lights[index]._isReady()) {
+                    return false;
+                }
+            }
+        }
+

```

Comment:
You should move this block before the call to `_prepareEffect`, as, conceptually, we wait for everything to be ready before creating the effect.

You should also replace `scene.lights` with `mesh.lightSources`, to avoid looping over lights that don't affect the mesh.